### ISSUE TITLE
canResolve does not work for submodules

### ISSUE THREAD

**Issue by ronawho** [chapel-lang/chapel#17234]
I want to check if a function in a submodule is available, but canResolve is not resolving the call.

```chpl
module MainModule {
  proc myproc() { writeln("myproc"); }
  module SubModule {
    proc myproc() { writeln("myproc submodule"); }
  }
}

module CanResolveTest {
  proc main () {
    use MainModule, Reflection;
    if canResolve("SubModule.myproc") {
      SubModule.myproc();
    } else {
      myproc();
    }
  }
}
```

Here `MainModule.myproc()` is called, but I was expecting `SubModule.myproc()` to be used

---

This is motivated by wanting to use `memoryUsed()` in a way that's compatible with 1.23 and 1.24, and `memoryUsed` was moved into a submodule during this release. e.g. I want to do something like:

```chpl
use Memory, Reflection;

var memUsed;
if canResolve("Diagnostics.memoryUsed") {
  memUsed = Diagnostics.memoryUsed();
} else {
  memUsed = memoryUsed();
}
writeln(memUsed);
```

**Comment by bradcray** [comment]
I agree that a feature like this would be useful, but am not surprised that it doesn't work today.  My interpretation of canResolve() as it's defined today is that it takes only a function name, not a more general expression that you might want to type into the code and call.  E.g., I don't think `myObject.foo()` would resolve either which is why there's a method variation that takes the object as a distinct argument.  From a design perspective, I'm not sure whether the "routine within a module" case should be handled by having the routine parse the function name to try and find, strip off, and resolve the module, or whether that should be passed as a separate (optional?) module path argument.
